homeassistant/components/habitica/config_flow.py

```diff
@@ -2,84 +2,184 @@
+                    session=session,
+                    username=user_input[CONF_USERNAME],
+                    password=user_input[CONF_PASSWORD],
+                )  # pyright: ignore[reportGeneralTypeIssues]
+
+            except ClientResponseError as ex:
+                if ex.status == HTTPStatus.UNAUTHORIZED:
+                    errors["base"] = "invalid_auth"
+                else:
+                    errors["base"] = "cannot_connect"
             except Exception:
                 _LOGGER.exception("Unexpected exception")
-                errors = {"base": "unknown"}
+                errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(info[CONF_API_USER])
+                await self.async_set_unique_id(login_response["id"])
                 self._abort_if_unique_id_configured()
-                return self.async_create_entry(title=info["title"], data=user_input)
+                return self.async_create_entry(
+                    title=login_response["username"],
+                    data={
+                        CONF_API_USER: login_response["id"],
+                        CONF_API_KEY: login_response["apiToken"],
+                        CONF_USERNAME: login_response["username"],
+                        CONF_URL: DEFAULT_URL,
+                        CONF_VERIFY_SSL: True,
+                    },
+                )
+
         return self.async_show_form(
-            step_id="user",
-            data_schema=DATA_SCHEMA,
+            step_id="login",
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=STEP_LOGIN_DATA_SCHEMA, suggested_values=user_input
+            ),
             errors=errors,
-            description_placeholders={},
         )
 
-    async def async_step_import(self, import_data):
+    async def async_step_advanced(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Advanced configuration with User Id and API Token.
+
+        Advanced configuration allows connecting to Habitica instances
+        hosted on different domains or to self-hosted instances.
+        """
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                session = async_get_clientsession(
+                    self.hass, verify_ssl=user_input.get(CONF_VERIFY_SSL, True)
+                )
+                api = await self.hass.async_add_executor_job(
+                    HabitipyAsync,
+                    {
+                        "login": user_input[CONF_API_USER],
+                        "password": user_input[CONF_API_KEY],
+                        "url": user_input.get(CONF_URL, DEFAULT_URL),
+                    },
+                )
+                api_response = await api.user.get(
+                    session=session,
+                    userFields="auth",
+                )  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
pyright?

Reply:
this PR is from May, i can't remember why I did that but my python skills have improved since 😂

```suggestion
                )  # type: ignore[reportGeneralTypeIssues]
```

Reply:
Ohhh, maybe because mypy complains about unused type: ignore 🤔

```

homeassistant/components/habitica/config_flow.py:100: error: Unused "type: ignore" comment  [unused-ignore]
homeassistant/components/habitica/config_flow.py:157: error: Unused "type: ignore" comment  [unused-ignore]
```